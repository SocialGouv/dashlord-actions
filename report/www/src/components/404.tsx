import * as React from "react";

import Table from "@codegouvfr/react-dsfr/Table";

import { Panel } from "./Panel";

type Report404Props = { data: Wget404Report };

export const Report404: React.FC<Report404Props> = ({ data }) => {
  const tableData = data && [
    ["destination"],
    ...data.map((url2) => [url2.link]),
  ];
  return (
    (data && data.length && (
      <Panel title="Erreurs 404" info="Pages introuvables">
        <Table data={tableData} />
      </Panel>
    )) ||
    null
  );
};
