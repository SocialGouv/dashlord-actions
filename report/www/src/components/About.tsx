import * as React from "react";

import CallOut from "@codegouvfr/react-dsfr/CallOut";

export const About: React.FC = () => (
  <CallOut className="fr-mb-3w" title="À propos">
    DashLord est <strong>100% open source</strong>, vous pouvez y contribuer en
    créant de la documentation, en proposant du code, ou encore en partageant
    vos questions{" "}
    <a
      href="https://github.com/socialgouv/dashlord"
      target="_blank"
      rel="noopener noreferrer"
    >
      sur le repository github
    </a>
    .
    <br /> <br />
    L&apos;objectif de DashLord est d&apos;identifier des points
    d&apos;attention mais aussi de proposer des solutions actionnables.
    <br /> <br />
    DashLord est né à la{" "}
    <a
      href="https://fabrique.social.gouv.fr"
      target="_blank"
      rel="noopener noreferrer"
    >
      {" "}
      Fabrique des ministères sociaux
    </a>{" "}
    pour répondre aux besoins d&apos;évaluation et de mise en oeuvre des bonnes
    pratiques de développement web.
    <br /> <br />
    Vous pouvez facilement créer votre propre DashLord pour vos URLs en suivant{" "}
    <a
      href="https://github.com/SocialGouv/dashlord/blob/main/README.md"
      target="_blank"
      rel="noopener noreferrer"
    >
      les instructions sur le GitHub
    </a>
    .
  </CallOut>
);
