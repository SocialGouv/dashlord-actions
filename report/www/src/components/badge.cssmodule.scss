.success,
.info,
.warning,
.danger {
    display: inline-block;
    cursor: pointer;
    padding: 4px 8px;
    color: white;
    min-width: 60px;
    border-radius: 4px;
    text-align: center;
    font-weight: bold;
}

.success {
    background-color: var(--success);
}

.info {
    background-color: var(--info);
}

.warning {
    background-color: #FF6F4C;
}

.danger {
    background-color: var(--error);
}