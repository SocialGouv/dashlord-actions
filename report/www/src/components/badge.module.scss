.success,
.info,
.warning,
.danger {
    display: inline-block;
    cursor: pointer;
    padding: 4px 8px;
    color: white;
    min-width: 60px;
    border-radius: 4px;
    text-align: center;
    font-weight: bold;
}



.success, .success:hover {
    background-color: var(--success) !important;
}

.info, .info:hover {
    background-color: var(--info) !important;
}

.warning, .warning:hover {
    background-color: var(--warning) !important;
}

.danger, .danger:hover {
    background-color: var(--error) !important;
}