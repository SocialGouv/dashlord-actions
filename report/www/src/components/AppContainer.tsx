import * as React from "react";

import { Container, Alert } from "@dataesr/react-dsfr";

import { Switch, Route, useLocation, useParams } from "react-router-dom";

import { Dashboard } from "./Dashboard";
import { Trends } from "./Trends";
import { Intro } from "./Intro";
import { About } from "./About";
import { WappalyzerDashboard } from "./WappalyzerDashboard";
import Url from "./Url";

const report: DashLordReport = require("./report.json");
const trends: Trends = require("./trends.json");

type CategoryRouteProps = { report: DashLordReport };

// for some reason react-router `:url*` didnt work, use `*` only
interface CategoryParamTypes {
  category: string;
}

const CategoryRoute: React.FC<CategoryRouteProps> = (props) => {
  const params = useParams<CategoryParamTypes>();
  const category = window.decodeURIComponent(params.category);
  const urls = props.report.filter((u) => u.category === category);
  return (
    <>
      <br />
      {urls.length ? (
        <>
          <h3>
            {category} :{urls.length} urls
          </h3>
          <Dashboard report={urls} />
        </>
      ) : (
        <h3>Aucun URL associ√©e</h3>
      )}
    </>
  );
};

interface TagParamTypes {
  tag: string;
}

type TagRouteProps = { report: DashLordReport };

const TagRoute: React.FC<TagRouteProps> = (props) => {
  const params = useParams<TagParamTypes>();
  const tag = window.decodeURIComponent(params.tag);
  const urls = props.report.filter((u) => u.tags && u.tags.includes(tag));
  return (
    <>
      <br />
      <h3>
        {tag} :{urls.length} urls
      </h3>
      <Dashboard report={urls} />
    </>
  );
};

interface UrlParamTypes {
  "0": string;
}

type UrlRouteProps = { report: DashLordReport };

const UrlRoute: React.FC<UrlRouteProps> = (props) => {
  const params = useParams<UrlParamTypes>();
  const url = window.decodeURIComponent(params["0"]);
  const urlData = props.report.find((u) => u.url === url);
  if (!urlData) {
    return (
      <Alert
        type="error"
        title={`Impossible de trouver le rapport pour ${url}`}
      />
    );
  }
  return <Url url={url} report={urlData} />;
};

export const AppContainer = () => {
  const location = useLocation();
  return (
    <Container fluid={location.pathname === "/"}>
      <div role="main" className="fr-my-4w">
        <Switch>
          <Route path="/url/*">
            <UrlRoute report={report} />
          </Route>
          <Route path="/dashboard">
            <Dashboard report={report} />
          </Route>
          <Route path="/trends">
            <Trends trends={trends} />
          </Route>
          <Route path="/category/:category">
            <CategoryRoute report={report} />
          </Route>
          <Route path="/tag/:tag">
            <TagRoute report={report} />
          </Route>
          <Route path="/wappalyzer">
            <WappalyzerDashboard report={report} />
          </Route>
          <Route path="/intro">
            <Intro />
          </Route>
          <Route path="/about">
            <About />
          </Route>
          <Route path="/">
            <Dashboard report={report} />
          </Route>
        </Switch>
      </div>
    </Container>
  );
};
