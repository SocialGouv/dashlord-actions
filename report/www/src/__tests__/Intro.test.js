import path from "path";
import * as React from "react";
import * as renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";

import { Intro } from "../components/Intro";

import report from "../report.json";

const mockSampleConfig = JSON.parse(
  jest
    .requireActual("fs")
    .readFileSync(path.join(__dirname, "config.json"))
    .toString()
);

// // prevent empty table due to resize detection
// jest.mock(
//   "react-virtualized-auto-sizer",
//   () =>
//     ({ children }) =>
//       children({ width: 1000, height: 1000 })
// );

it("Should render Intro", () => {
  const props = {report};
  const tree = renderer.create(<MemoryRouter><Intro {...props} /></MemoryRouter>).toJSON();
  expect(tree).toMatchSnapshot();
});

 
describe("Tools config", () => {
  beforeEach(() => {
    jest.mock("../config.json", () => ({
      ...mockSampleConfig,
      tools: { http: false, zap: true, lighthouse: false },
    }));
  });

  it("Should render Intro with limited tools", () => {
    const props = { report };
    const tree = renderer
      .create(
        <MemoryRouter>
          <Intro {...props} />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
