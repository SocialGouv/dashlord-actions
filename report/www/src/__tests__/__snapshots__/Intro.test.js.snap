// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Intro 1`] = `
<div>
  <div
    class="fr-callout fr-mb-3w"
  >
    <h1
      class="fr-callout__title fr-text--lg"
    >
      DashLord Fabrique des ministères sociaux
    </h1>
    <p
      class="fr-callout__text fr-text--lg"
    >
      <br />
      DashLord compile les données techniques issues de différents outils open-source.
      <br />
      Cette évaluation ne remplace en aucun cas une expertise manuelle, et seule la page d'accueil du site est ici évaluée.
      <br />
      Vous pouvez
       
      <a
        href="https://github.com/socialgouv/dashlord/issues/new"
        rel="noopener noreferrer"
        target="_blank"
      >
        soumettre de nouvelles URLs, proposer des corrections
      </a>
       
      ou
       
      <a
        href="https://github.com/orgs/SocialGouv/projects/13"
        rel="noopener noreferrer"
        target="_blank"
      >
        consulter la roadmap
      </a>
      .
      <br />
      <br />
    </p>
    <button
      class="fr-btn--md fr-btn"
      type="button"
    >
      <a
        href="/"
      >
        Accéder au tableau de bord
      </a>
    </button>
  </div>
  <div
    class="container"
  >
    <div
      class="banner"
    >
      <h5
        class="mainTitle"
      >
        Google Lighthouse
      </h5>
      <a
        href="https://developers.google.com/web/tools/lighthouse"
        rel="noreferrer noopener"
        style="float: right;"
        target="_blank"
      >
        Site officiel
      </a>
    </div>
    <div
      class="body"
    >
      Permet un audit automatique de page web :
      <br />
      <br />
      <li>
        Performances de chargement
      </li>
      <li>
        Bonnes pratiques SEO (référencement)
      </li>
      <li>
        Bonnes pratiques accessibilité (moteur Axe)
      </li>
      <li>
        Bonnes pratiques web
      </li>
      <br />
      <div
        class="fr-alert fr-alert--success"
      >
        <p
          class="fr-alert__title"
        />
        <p>
          L'audit complet avec les recommandations de correction est disponible pour chaque URL
        </p>
      </div>
      <div
        class="fr-alert fr-alert--info"
      >
        <p
          class="fr-alert__title"
        />
        <p>
          Les mesures de performance sont un instantané, donc pas forcément représentatives
        </p>
      </div>
      <div
        class="fr-alert fr-alert--error"
      >
        <p
          class="fr-alert__title"
        />
        <p>
          <span>
            Les mesures d'accessibilité n'abordent pas du tout
             
            <a
              href="https://www.numerique.gouv.fr/publications/rgaa-accessibilite/"
              rel="noopener noreferrer"
              target="_blank"
            >
              le RGAA
            </a>
            .
          </span>
        </p>
      </div>
    </div>
  </div>
  <div
    class="container"
  >
    <div
      class="banner"
    >
      <h5
        class="mainTitle"
      >
        Dependabot
      </h5>
      <a
        href="https://dependabot.com/"
        rel="noreferrer noopener"
        style="float: right;"
        target="_blank"
      >
        Site officiel
      </a>
    </div>
    <div
      class="body"
    >
      Recense les vulnérabilités de tes dépendances dans le code source
      <br />
      <br />
      <a
        href="https://docs.github.com/en/code-security/supply-chain-security/about-alerts-for-vulnerable-dependencies"
        rel="noopener noreferrer"
        target="_blank"
      >
        Scan des vulnérabilités des dépendances sur le dépôt Github du code
      </a>
      <br />
      <br />
      <div
        class="fr-alert fr-alert--info"
      >
        <p
          class="fr-alert__title"
        />
        <p>
          Nécessite d'avoir activé Dependabot sur ses repositories GitHub
        </p>
      </div>
    </div>
  </div>
  <div
    class="container"
  >
    <div
      class="banner"
    >
      <h5
        class="mainTitle"
      >
        Codescan
      </h5>
      <a
        href="https://docs.github.com/en/code-security/secure-coding/about-code-scanning"
        rel="noreferrer noopener"
        style="float: right;"
        target="_blank"
      >
        Documentation
      </a>
    </div>
    <div
      class="body"
    >
      Recense les potentielles vulnérabilités dans ton code
      <br />
      <br />
      <li>
        Scan des potentielles vulnérabilités sur le dépôt Github du code
      </li>
      <li>
        Possibilité d'activer autant de scanner souhaité: CodeQL, etc
      </li>
      <li>
        Liste restreinte de langages couverts: C/C++, C#, Go, Java, JavaScript/TypeScript, Python
      </li>
       
      <br />
      <br />
      <div
        class="fr-alert fr-alert--info"
      >
        <p
          class="fr-alert__title"
        />
        <p>
          Nécessite d'avoir activé CodeQL sur le repository
        </p>
      </div>
    </div>
  </div>
  <div
    class="container"
  >
    <div
      class="banner"
    >
      <h5
        class="mainTitle"
      >
        Nmap
      </h5>
      <a
        href="https://nmap.org"
        rel="noreferrer noopener"
        style="float: right;"
        target="_blank"
      >
        Site officiel
      </a>
    </div>
    <div
      class="body"
    >
      Nmap scan les vulnérabilités d'une machine (IP) associée à un domaine
      <br />
      <br />
      <a
        href="https://nmap.org/nsedoc/scripts/vulners.html"
        rel="noopener noreferrer"
        target="_blank"
      >
        Scan des ports ouverts avec vulnérabilités grâce au script vulners
      </a>
    </div>
  </div>
  <div
    class="container"
  >
    <div
      class="banner"
    >
      <h5
        class="mainTitle"
      >
        OWASP Zed Attack Proxy
      </h5>
      <a
        href="https://www.zaproxy.org/docs/docker/baseline-scan/"
        rel="noreferrer noopener"
        style="float: right;"
        target="_blank"
      >
        Site officiel
      </a>
    </div>
    <div
      class="body"
    >
      Scan de vulnérabilités passif "baseline" qui permet de détecter des risques de sécurité.
      <br />
      <br />
      <li>
        Bonnes pratiques web
      </li>
      <li>
        Bonnes pratiques http
      </li>
      <br />
      <div
        class="fr-alert fr-alert--info"
      >
        <p
          class="fr-alert__title"
        />
        <p>
          L'audit complet avec les recommandations de correction est disponible pour chaque URL
        </p>
      </div>
    </div>
  </div>
  <div
    class="container"
  >
    <div
      class="banner"
    >
      <h5
        class="mainTitle"
      >
        testssl.sh
      </h5>
      <svg
        fill="none"
        height="20"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        style="margin-left: 5px; cursor: pointer;"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="10"
        />
        <line
          x1="12"
          x2="12"
          y1="16"
          y2="12"
        />
        <line
          x1="12"
          x2="12.01"
          y1="8"
          y2="8"
        />
      </svg>
      <a
        href="https://testssl.sh/"
        rel="noreferrer noopener"
        style="float: right;"
        target="_blank"
      >
        Site officiel
      </a>
    </div>
    <div
      class="body"
    >
      Évalue le niveau de confiance d'un certificat SSL
      <br />
      <br />
      <li>
        Bonnes pratiques de configuration
      </li>
      <li>
        Protocoles disponibles
      </li>
      <li>
        Compatibilité navigateurs
      </li>
      <li>
        Solidité des clés de chiffrement
      </li>
      <br />
      <div
        class="fr-alert fr-alert--info"
      >
        <p
          class="fr-alert__title"
        />
        <p>
          L'audit complet avec les recommandations de correction est disponible pour chaque URL
        </p>
      </div>
    </div>
  </div>
  <div
    class="container"
  >
    <div
      class="banner"
    >
      <h5
        class="mainTitle"
      >
        Mozilla HTTP observatory
      </h5>
      <svg
        fill="none"
        height="20"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        style="margin-left: 5px; cursor: pointer;"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="10"
        />
        <line
          x1="12"
          x2="12"
          y1="16"
          y2="12"
        />
        <line
          x1="12"
          x2="12.01"
          y1="8"
          y2="8"
        />
      </svg>
      <a
        href="https://observatory.mozilla.org/"
        rel="noreferrer noopener"
        style="float: right;"
        target="_blank"
      >
        Site officiel
      </a>
    </div>
    <div
      class="body"
    >
      Évalue le niveau de qualité/sécurité de la page web et de son serveur
      <br />
      <br />
      <li>
        Bonnes pratiques de configuration
      </li>
      <li>
        Bonnes pratiques web
      </li>
      <li>
        Solidité des clés de chiffrement
      </li>
    </div>
  </div>
  <div
    class="container"
  >
    <div
      class="banner"
    >
      <h5
        class="mainTitle"
      >
        Updown.io
      </h5>
      <svg
        fill="none"
        height="20"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        style="margin-left: 5px; cursor: pointer;"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="10"
        />
        <line
          x1="12"
          x2="12"
          y1="16"
          y2="12"
        />
        <line
          x1="12"
          x2="12.01"
          y1="8"
          y2="8"
        />
      </svg>
      <a
        href="https://updown.io/"
        rel="noreferrer noopener"
        style="float: right;"
        target="_blank"
      >
        Site officiel
      </a>
    </div>
    <div
      class="body"
    >
      Évalue la qualité de service rendue par le serveur
      <br />
      <br />
      <li>
        Temps de réponse
      </li>
      <li>
        Disponibilité
      </li>
      <li>
        Indice de performance : APDEX
      </li>
      <li>
        Validité des certificats TLS
      </li>
      <br />
      <div
        class="fr-alert fr-alert--info"
      >
        <p
          class="fr-alert__title"
        />
        <p>
          Vous devez disposer d'un compte updown.io et d'une clé API
        </p>
      </div>
    </div>
  </div>
  <div
    class="container"
  >
    <div
      class="banner"
    >
      <h5
        class="mainTitle"
      >
        Nucléi
      </h5>
      <a
        href="https://nuclei.projectdiscovery.io/"
        rel="noreferrer noopener"
        style="float: right;"
        target="_blank"
      >
        Site officiel
      </a>
    </div>
    <div
      class="body"
    >
      Détecte plus de 700 erreurs de configuration courantes sur les applications webs.
      <br />
      <br />
      <li>
        Sécurité
      </li>
      <li>
        Bonnes pratiques web
      </li>
    </div>
  </div>
  <div
    class="container"
  >
    <div
      class="banner"
    >
      <h5
        class="mainTitle"
      >
        Third-parties
      </h5>
      <a
        href="https://github.com/SocialGouv/thirdparties"
        rel="noreferrer noopener"
        style="float: right;"
        target="_blank"
      >
        Code source
      </a>
    </div>
    <div
      class="body"
    >
      Liste tous les scripts externes chargés par une URL et qui peuvent avoir un impact sur :
      <br />
      <br />
      <li>
        Performances web
      </li>
      <li>
        Sécurité
      </li>
      <li>
        Vie privée
      </li>
      <br />
      <div
        class="fr-alert fr-alert--error"
      >
        <p
          class="fr-alert__title"
        >
          thirdparties
        </p>
        <p>
          Certains script légitimes peuvent apparaitre dans cette catégorie s'ils sont hébergés sur d'autres serveurs
        </p>
      </div>
    </div>
  </div>
  <div
    class="container"
  >
    <div
      class="banner"
    >
      <h5
        class="mainTitle"
      >
        GeoIP2
      </h5>
      <a
        href="https://www.maxmind.com/en/geoip-demo"
        rel="noreferrer noopener"
        style="float: right;"
        target="_blank"
      >
        Site officiel
      </a>
    </div>
    <div
      class="body"
    >
      Géolocalise tous les serveurs contactés lors de l'ouverture d'une URL.
      <br />
      <br />
      <li>
        Vie privée
      </li>
    </div>
  </div>
  <div
    class="container"
  >
    <div
      class="banner"
    >
      <h5
        class="mainTitle"
      >
        Wappalyzer
      </h5>
      <a
        href="https://www.wappalyzer.com/"
        rel="noreferrer noopener"
        style="float: right;"
        target="_blank"
      >
        Site officiel
      </a>
    </div>
    <div
      class="body"
    >
      Wappalyzer reconnait +1500 technologies web, Javascript, CMS, outillage...
      <br />
      <br />
      <li>
        Stack technique
      </li>
      <li>
        Obsolescence
      </li>
      <li>
        Parc
      </li>
    </div>
  </div>
  <div
    class="container"
  >
    <div
      class="banner"
    >
      <h5
        class="mainTitle"
      >
        Statistiques
      </h5>
    </div>
    <div
      class="body"
    >
      Vérifie si la page /stats existe
      <br />
      <br />
      Par exemple:
       
      <a
        href="https://beta.gouv.fr/stats"
        rel="noopener noreferrer"
        target="_blank"
      >
        Beta.gouv.fr
      </a>
    </div>
  </div>
  <div
    class="container"
  >
    <div
      class="banner"
    >
      <h5
        class="mainTitle"
      >
        Erreurs 404
      </h5>
    </div>
    <div
      class="body"
    >
      Explore le site web et détecte les liens brisés.
    </div>
  </div>
  <div
    class="container"
  >
    <div
      class="banner"
    >
      <h5
        class="mainTitle"
      >
        Vulnérabilités Trivy
      </h5>
      <a
        href="https://aquasecurity.github.io/trivy/"
        rel="noreferrer noopener"
        style="float: right;"
        target="_blank"
      >
        Site officiel
      </a>
    </div>
    <div
      class="body"
    >
      Trivy permet de scanner les containers docker à la recherche de vulnérabilités connues.
      <br />
      <br />
      Trivy détecte les vulnérabilités niveau OS (Alpine, RHEL, CentOS, etc.) mais aussi niveau packages (Bundler, Composer, npm, yarn, etc.).
    </div>
  </div>
  <div
    class="container"
  >
    <div
      class="banner"
    >
      <h5
        class="mainTitle"
      >
        Score eco-index
      </h5>
      <a
        href="https://www.ecoindex.fr/comment-ca-marche/"
        rel="noreferrer noopener"
        style="float: right;"
        target="_blank"
      >
        Comment ça marche
      </a>
    </div>
    <div
      class="body"
    >
      EcoIndex est un outil communautaire, gratuit et transparent qui, pour une URL donnée, permet d’évaluer :
      <br />
      <br />
      <li>
        sa performance environnementale absolue à l’aide d’un score sur 100 (higher is better)
      </li>
      <li>
        sa performance environnementale relative à l’aide d’une note de A à G
      </li>
      <li>
        l’empreinte technique de la page (poids, complexité, etc.)
      </li>
      <li>
        l’empreinte environnementale associée (gaz à effet de serre et eau).
      </li>
      <br />
      L’objectif d’EcoIndex est d’aider le plus grand nombre à prendre conscience de l’impact environnemental de l’internet.
    </div>
  </div>
  <div
    class="container"
  >
    <div
      class="banner"
    >
      <h5
        class="mainTitle"
      >
        Score SonarCloud
      </h5>
      <a
        href="https://sonarcloud.io/"
        rel="noreferrer noopener"
        style="float: right;"
        target="_blank"
      >
        Site officiel
      </a>
    </div>
    <div
      class="body"
    >
      SonarCloud permet d'analyser le code source des repositories et de détecter des bugs, vulnérabilités, duplications et autres indicateurs de qualité.
    </div>
  </div>
  <div
    class="container"
  >
    <div
      class="banner"
    >
      <h5
        class="mainTitle"
      >
        Détection du système de design de l'état
      </h5>
      <a
        href="https://www.systeme-de-design.gouv.fr/"
        rel="noreferrer noopener"
        style="float: right;"
        target="_blank"
      >
        Site officiel
      </a>
    </div>
    <div
      class="body"
    >
      Détection du système de design de l'état avec la balise
      <code>
        fr-header__brand
      </code>
    </div>
  </div>
</div>
`;

exports[`Tools config Should render Intro with limited tools 1`] = `
<div>
  <div
    class="fr-callout fr-mb-3w"
  >
    <h1
      class="fr-callout__title fr-text--lg"
    >
      DashLord Fabrique des ministères sociaux
    </h1>
    <p
      class="fr-callout__text fr-text--lg"
    >
      <br />
      DashLord compile les données techniques issues de différents outils open-source.
      <br />
      Cette évaluation ne remplace en aucun cas une expertise manuelle, et seule la page d'accueil du site est ici évaluée.
      <br />
      Vous pouvez
       
      <a
        href="https://github.com/socialgouv/dashlord/issues/new"
        rel="noopener noreferrer"
        target="_blank"
      >
        soumettre de nouvelles URLs, proposer des corrections
      </a>
       
      ou
       
      <a
        href="https://github.com/orgs/SocialGouv/projects/13"
        rel="noopener noreferrer"
        target="_blank"
      >
        consulter la roadmap
      </a>
      .
      <br />
      <br />
    </p>
    <button
      class="fr-btn--md fr-btn"
      type="button"
    >
      <a
        href="/"
      >
        Accéder au tableau de bord
      </a>
    </button>
  </div>
  <div
    class="container"
  >
    <div
      class="banner"
    >
      <h5
        class="mainTitle"
      >
        Dependabot
      </h5>
      <a
        href="https://dependabot.com/"
        rel="noreferrer noopener"
        style="float: right;"
        target="_blank"
      >
        Site officiel
      </a>
    </div>
    <div
      class="body"
    >
      Recense les vulnérabilités de tes dépendances dans le code source
      <br />
      <br />
      <a
        href="https://docs.github.com/en/code-security/supply-chain-security/about-alerts-for-vulnerable-dependencies"
        rel="noopener noreferrer"
        target="_blank"
      >
        Scan des vulnérabilités des dépendances sur le dépôt Github du code
      </a>
      <br />
      <br />
      <div
        class="fr-alert fr-alert--info"
      >
        <p
          class="fr-alert__title"
        />
        <p>
          Nécessite d'avoir activé Dependabot sur ses repositories GitHub
        </p>
      </div>
    </div>
  </div>
  <div
    class="container"
  >
    <div
      class="banner"
    >
      <h5
        class="mainTitle"
      >
        Codescan
      </h5>
      <a
        href="https://docs.github.com/en/code-security/secure-coding/about-code-scanning"
        rel="noreferrer noopener"
        style="float: right;"
        target="_blank"
      >
        Documentation
      </a>
    </div>
    <div
      class="body"
    >
      Recense les potentielles vulnérabilités dans ton code
      <br />
      <br />
      <li>
        Scan des potentielles vulnérabilités sur le dépôt Github du code
      </li>
      <li>
        Possibilité d'activer autant de scanner souhaité: CodeQL, etc
      </li>
      <li>
        Liste restreinte de langages couverts: C/C++, C#, Go, Java, JavaScript/TypeScript, Python
      </li>
       
      <br />
      <br />
      <div
        class="fr-alert fr-alert--info"
      >
        <p
          class="fr-alert__title"
        />
        <p>
          Nécessite d'avoir activé CodeQL sur le repository
        </p>
      </div>
    </div>
  </div>
  <div
    class="container"
  >
    <div
      class="banner"
    >
      <h5
        class="mainTitle"
      >
        Nmap
      </h5>
      <a
        href="https://nmap.org"
        rel="noreferrer noopener"
        style="float: right;"
        target="_blank"
      >
        Site officiel
      </a>
    </div>
    <div
      class="body"
    >
      Nmap scan les vulnérabilités d'une machine (IP) associée à un domaine
      <br />
      <br />
      <a
        href="https://nmap.org/nsedoc/scripts/vulners.html"
        rel="noopener noreferrer"
        target="_blank"
      >
        Scan des ports ouverts avec vulnérabilités grâce au script vulners
      </a>
    </div>
  </div>
  <div
    class="container"
  >
    <div
      class="banner"
    >
      <h5
        class="mainTitle"
      >
        OWASP Zed Attack Proxy
      </h5>
      <a
        href="https://www.zaproxy.org/docs/docker/baseline-scan/"
        rel="noreferrer noopener"
        style="float: right;"
        target="_blank"
      >
        Site officiel
      </a>
    </div>
    <div
      class="body"
    >
      Scan de vulnérabilités passif "baseline" qui permet de détecter des risques de sécurité.
      <br />
      <br />
      <li>
        Bonnes pratiques web
      </li>
      <li>
        Bonnes pratiques http
      </li>
      <br />
      <div
        class="fr-alert fr-alert--info"
      >
        <p
          class="fr-alert__title"
        />
        <p>
          L'audit complet avec les recommandations de correction est disponible pour chaque URL
        </p>
      </div>
    </div>
  </div>
  <div
    class="container"
  >
    <div
      class="banner"
    >
      <h5
        class="mainTitle"
      >
        testssl.sh
      </h5>
      <svg
        fill="none"
        height="20"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        style="margin-left: 5px; cursor: pointer;"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="10"
        />
        <line
          x1="12"
          x2="12"
          y1="16"
          y2="12"
        />
        <line
          x1="12"
          x2="12.01"
          y1="8"
          y2="8"
        />
      </svg>
      <a
        href="https://testssl.sh/"
        rel="noreferrer noopener"
        style="float: right;"
        target="_blank"
      >
        Site officiel
      </a>
    </div>
    <div
      class="body"
    >
      Évalue le niveau de confiance d'un certificat SSL
      <br />
      <br />
      <li>
        Bonnes pratiques de configuration
      </li>
      <li>
        Protocoles disponibles
      </li>
      <li>
        Compatibilité navigateurs
      </li>
      <li>
        Solidité des clés de chiffrement
      </li>
      <br />
      <div
        class="fr-alert fr-alert--info"
      >
        <p
          class="fr-alert__title"
        />
        <p>
          L'audit complet avec les recommandations de correction est disponible pour chaque URL
        </p>
      </div>
    </div>
  </div>
  <div
    class="container"
  >
    <div
      class="banner"
    >
      <h5
        class="mainTitle"
      >
        Updown.io
      </h5>
      <svg
        fill="none"
        height="20"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        style="margin-left: 5px; cursor: pointer;"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="10"
        />
        <line
          x1="12"
          x2="12"
          y1="16"
          y2="12"
        />
        <line
          x1="12"
          x2="12.01"
          y1="8"
          y2="8"
        />
      </svg>
      <a
        href="https://updown.io/"
        rel="noreferrer noopener"
        style="float: right;"
        target="_blank"
      >
        Site officiel
      </a>
    </div>
    <div
      class="body"
    >
      Évalue la qualité de service rendue par le serveur
      <br />
      <br />
      <li>
        Temps de réponse
      </li>
      <li>
        Disponibilité
      </li>
      <li>
        Indice de performance : APDEX
      </li>
      <li>
        Validité des certificats TLS
      </li>
      <br />
      <div
        class="fr-alert fr-alert--info"
      >
        <p
          class="fr-alert__title"
        />
        <p>
          Vous devez disposer d'un compte updown.io et d'une clé API
        </p>
      </div>
    </div>
  </div>
  <div
    class="container"
  >
    <div
      class="banner"
    >
      <h5
        class="mainTitle"
      >
        Nucléi
      </h5>
      <a
        href="https://nuclei.projectdiscovery.io/"
        rel="noreferrer noopener"
        style="float: right;"
        target="_blank"
      >
        Site officiel
      </a>
    </div>
    <div
      class="body"
    >
      Détecte plus de 700 erreurs de configuration courantes sur les applications webs.
      <br />
      <br />
      <li>
        Sécurité
      </li>
      <li>
        Bonnes pratiques web
      </li>
    </div>
  </div>
  <div
    class="container"
  >
    <div
      class="banner"
    >
      <h5
        class="mainTitle"
      >
        Third-parties
      </h5>
      <a
        href="https://github.com/SocialGouv/thirdparties"
        rel="noreferrer noopener"
        style="float: right;"
        target="_blank"
      >
        Code source
      </a>
    </div>
    <div
      class="body"
    >
      Liste tous les scripts externes chargés par une URL et qui peuvent avoir un impact sur :
      <br />
      <br />
      <li>
        Performances web
      </li>
      <li>
        Sécurité
      </li>
      <li>
        Vie privée
      </li>
      <br />
      <div
        class="fr-alert fr-alert--error"
      >
        <p
          class="fr-alert__title"
        >
          thirdparties
        </p>
        <p>
          Certains script légitimes peuvent apparaitre dans cette catégorie s'ils sont hébergés sur d'autres serveurs
        </p>
      </div>
    </div>
  </div>
  <div
    class="container"
  >
    <div
      class="banner"
    >
      <h5
        class="mainTitle"
      >
        GeoIP2
      </h5>
      <a
        href="https://www.maxmind.com/en/geoip-demo"
        rel="noreferrer noopener"
        style="float: right;"
        target="_blank"
      >
        Site officiel
      </a>
    </div>
    <div
      class="body"
    >
      Géolocalise tous les serveurs contactés lors de l'ouverture d'une URL.
      <br />
      <br />
      <li>
        Vie privée
      </li>
    </div>
  </div>
  <div
    class="container"
  >
    <div
      class="banner"
    >
      <h5
        class="mainTitle"
      >
        Wappalyzer
      </h5>
      <a
        href="https://www.wappalyzer.com/"
        rel="noreferrer noopener"
        style="float: right;"
        target="_blank"
      >
        Site officiel
      </a>
    </div>
    <div
      class="body"
    >
      Wappalyzer reconnait +1500 technologies web, Javascript, CMS, outillage...
      <br />
      <br />
      <li>
        Stack technique
      </li>
      <li>
        Obsolescence
      </li>
      <li>
        Parc
      </li>
    </div>
  </div>
  <div
    class="container"
  >
    <div
      class="banner"
    >
      <h5
        class="mainTitle"
      >
        Erreurs 404
      </h5>
    </div>
    <div
      class="body"
    >
      Explore le site web et détecte les liens brisés.
    </div>
  </div>
</div>
`;
